packages/agent/src/actor.ts

```diff
@@ -530,25 +530,41 @@ function _createActorMethod(
         arg,
         effectiveCanisterId: ecid,
       });
-
-      requestId;
-      response;
-      requestDetails;
-
-      if (!response.ok || response.body /* IC-1462 */) {
-        throw new UpdateCallRejectedError(cid, methodName, requestId, response);
+      let reply: ArrayBuffer | undefined;
+      let certificate: Certificate | undefined;
+      if (response.body && response.body.certificate) {
+        const cert = response.body.certificate;
+        certificate = await Certificate.create({
```

Comment:
We can check if agent.rootKey is initialized before verifying as done in `pollForResponse`